hooks/tk-flame_actions.py

```diff
@@ -0,0 +1,383 @@
+    # methods called by the menu options in the loader
+
+    def _import_batch_file(self, sg_publish_data):
+        """
+        Imports a Batch Group from Shotgun into Flame.
+        
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :type sg_publish_data: dict
+        """
+
+        batch_path = self.get_publish_path(sg_publish_data)
+
+        # Only load the batch if it exists
+        if batch_path and os.path.exists(batch_path):
+            flame.batch.go_to()
+            flame.batch.load_setup(batch_path)
+            flame.batch.organize()
+        else:
+            raise FlameActionError("File not found on disk - '%s'" % batch_path)
+
+    def _import_clip(self, sg_publish_data):
+        """
+        Imports a clip to the current Batch Group.
+
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :type sg_publish_data: dict
+        """
+
+        clip_path = self.get_publish_path(sg_publish_data)
+
+        # Only load the clip if it exists
+        if clip_path and os.path.exists(clip_path):
+            flame.batch.go_to()
+            flame.batch.import_clip(clip_path, SCHEMATIC_REEL)
+            flame.batch.organize()
+        elif '%' in clip_path:
+            try:
+                # Special case parsing for frames, attempts to expand
+                temp_filters = [["id", "is", sg_publish_data["version"]["id"]]]
+                temp_fields = ["frame_range"]
+                temp_type = "Version"
+
+                temp_info = self.parent.shotgun.find_one(
+                    temp_type, filters=temp_filters, fields=temp_fields
+                )
+
+                # Unfortunately, we can't do simple formatting here as %<num>d
+                # old style Python formatting does not support getting a frame
+                # range. Thus we need to parse it ourselves
+                new_path = self._handle_frame_range(
+                    clip_path, temp_info["frame_range"]
+                )
+
+                if new_path and len(new_path) != 0:
+                    clip_path = new_path
+
+                flame.batch.go_to()
+                flame.batch.import_clip(clip_path, SCHEMATIC_REEL)
+                flame.batch.organize()
+            except Exception:
+                raise FlameActionError("File not found on disk - '%s'" % clip_path)
+        else:
+            raise FlameActionError("File not found on disk - '%s'" % clip_path)
+
+    def _import_batch_group(self, sg_publish_data):
+        """
+        Imports a Batch Group from Shotgun into Flame.
+
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :type sg_publish_data: dict
+        """
+        # Determines published files
+        sg_filters = [["id", "is", sg_publish_data["id"]]]
+        sg_fields = ["sg_published_files",
```

Comment:
I'm worried here. The `sg_` are not part of the `PublishedFile` entity in Shotgun. Are those fields you have added on your site? Clients will not have these fields and running this hook will fail. We should grab some time offline to discuss this. Please ping me on Slack so we can schedule something.